### PR TITLE
contact form

### LINKED ISSUE #184: General Contactform
- Name*
- E-Mail*
- Telephone
- Image Attachement
- Subject*
  - ... tbd 
- Message*

### PR DESCRIPTION
**Pull request by BoyPersoon**
Hi Max,

Could you take a look at this!? My PHP is a bit rusty so let me know if you have any suggestions!
Also we would need some text in German, and in general at the top of the form to explain it's purpose.

### CODE REVIEW

File: src/Infrastructure/Controller/ContactController.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            $file = $request->files->get('file');
+
+            $mailBody = $this->twig->render(
+                '/email/contact_form.html.twig',
+                $params);
+
+            $email = new Email();
+            $email->subject($params['subject']);
+            $email->html($mailBody);
+            $email->to($params['email']);
```
Comment (bartrail) [inline src/Infrastructure/Controller/ContactController.php]:
```suggestion
            $email->to('contact@print4health.org');
            $email->cc($params['email']);
```
I would make it public that the email was sent to us and then is cc'ed to the sender too.

File: src/Infrastructure/Controller/ContactController.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            $email->bcc("contact@print4health.org");
```
Comment (bartrail) [inline src/Infrastructure/Controller/ContactController.php]:
```suggestion
```

File: .gitignore [hunk 3]
```diff
@@ -31,3 +31,6 @@ yarn-error.log
 
 cypress/screenshots
 cypress/videos
+
+###> jetbrains editors ###
+/.idea
```
Comment (gherkins) [inline .gitignore]: i think that would rather belong in your local `.git/info/exclude`

File: assets/js/app.js [hunk 4]
```diff
@@ -136,6 +137,9 @@ class App extends React.Component {
                     <li className="nav-item">
                       <NavLink className="nav-link" activeClassName="text-primary" to="/faq">FAQ</NavLink>
                     </li>
+                    <li className="nav-item">
```
Comment (gherkins) [inline assets/js/app.js]: we need to doublecheck that theres enough space left in the main-navbar

maybe move this to the footer? @bartrail

File: src/Infrastructure/Controller/ContactController.php [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    public function formDataAction(
+        Request $request,
+        ContactMailer $contactMailer
+    ): JsonResponse {
+        try {
+            $params = $request->request->all();
+            $file = $request->files->get('file');
+            $params['filePath'] = null;
+            $params['file'] = null;
+
```
Comment (gherkins) [inline src/Infrastructure/Controller/ContactController.php:75]: @bartrail  shouldn't we add more server-side validation, as in the registration form?

File: src/Domain/Contact/Mailer.php [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+class Mailer
+{
+    private MailerInterface $mailer;
+    private Environment $twig;
+
+    public function __construct(
+        MailerInterface $mailer,
+        Environment $twig
+    ) {
```
Comment (bartrail) [inline src/Domain/Contact/Mailer.php:24]: please add `string $from` as argument so we use the global .env variable (see the other mailer and the `services.yaml` for how to configure the env variables

File: src/Domain/Contact/Mailer.php [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        $mailBody = $this->twig->render(
+            '/email/contact_form.html.twig',
+            $params);
+
+        $email = new Email();
+        $email->subject($params['subject'])
+            ->html($mailBody)
+            ->cc($params['email'])
+            ->to('contact@print4health.org')
+            ->from('contact@print4health.org')
```
Comment (bartrail) [inline src/Domain/Contact/Mailer.php]: the `$from` variable can then be used here

**Comment by bartrail** [comment]
we had a discussion whether we should persist the contact form also in database for further processing. What do you think about it?

**Comment by BoyPersoon** [comment]
I think we should prioritize getting the contact form online and add that functionality later (as a different issue).

Adding text on top of the form is more important right now I think.

**Review by bartrail — DISMISSED** [review]

**Review by gherkins — CHANGES_REQUESTED** [review]
sorry, i meant to submit this earlier, but somehow didn't 🤔

**Review by gherkins — APPROVED** [review]

**Pull request merged**